Description: FindImageMagick: Use find_program instead of find_path
 Since cmake 3.6 find_path doesn't use the system search paths anymore.
 As a result FindImageMagick wasn't able to get the executable dir.
Author: Felix Geyer <fgeyer@debian.org>
Forwarded: https://gitlab.kitware.com/cmake/cmake/merge_requests/30

--- a/Modules/FindImageMagick.cmake
+++ b/Modules/FindImageMagick.cmake
@@ -185,17 +185,21 @@ endfunction()
 # Start Actual Work
 #---------------------------------------------------------------------
 # Try to find a ImageMagick installation binary path.
-find_path(ImageMagick_EXECUTABLE_DIR
+find_program(ImageMagick_EXECUTABLE_FILENAME
   NAMES mogrify${CMAKE_EXECUTABLE_SUFFIX}
   PATHS
     "[HKEY_LOCAL_MACHINE\\SOFTWARE\\ImageMagick\\Current;BinPath]"
   DOC "Path to the ImageMagick binary directory."
   NO_DEFAULT_PATH
   )
-find_path(ImageMagick_EXECUTABLE_DIR
+find_program(ImageMagick_EXECUTABLE_FILENAME
   NAMES mogrify${CMAKE_EXECUTABLE_SUFFIX}
   )
 
+if(ImageMagick_EXECUTABLE_FILENAME)
+  get_filename_component(ImageMagick_EXECUTABLE_DIR "${ImageMagick_EXECUTABLE_FILENAME}" DIRECTORY)
+endif()
+
 # Find each component. Search for all tools in same dir
 # <ImageMagick_EXECUTABLE_DIR>; otherwise they should be found
 # independently and not in a cohesive module such as this one.
